src/en/news/blog/2021/managing-grafana-dashboards-with-grafonnet/index.md

```diff
@@ -0,0 +1,94 @@
+---
+title: "Managing Grafana Dashboards with Grafonnet"
+date: "2021-09-17"
+author: "Aashish Sharma"
+---
+
+The process of maintaing grafana dashboards or updating the dashboard JSON files can sometimes become a mess as the size of the JSON files becomes too large to debug it. In this blog you can find out how grafana dashboards can be managed as code using Grafonnet.
+
+## About Grafonnet
+
+In case you already know what [Grafonnet](https://github.com/grafana/grafonnet-lib) is, it would be easier for you to get going with grafonnet. For those who don’t know: Grafonnet is a [Jsonnet](https://jsonnet.org) library – with Jsonnet being a JSON extension to create files using abstractions such as variables, functions, conditionals and more.Grafonnet consists of different functions we can use to create the JSON objects required to compose Grafana dashboards or panels. the resulting JSON files are then imported into grafana.
+
+## Creating a Dashboard with Grafonnet
+
+The grafonnet library exposes functions we can use to define our Grafana dashboards. First, we need to [import](https://grafana.github.io/grafonnet-lib/getting-started/) Grafonnet. Grafana.libsonnet works as an entry point and exposes all required functions and interfaces from the library files.
+The code below is the minimal code required to create a dashboard requiring only a title.
+
+```
+local grafana = import 'grafonnet/grafana.libsonnet';
+local dashboard = grafana.dashboard;
+ 
+dashboard.new(
+  title='Demo dashboard'
+)
+```
+
+The above code can be compiled with `$ jsonnet -o new.json new.jsonnet` and the resultant json file(new.json) can be imported into Grafana.
+
+## Adding Template Variables
+
+Grafonnet library provides the functions `addTemplate() and addTemplates()`, which either take a template object or an array of them.
+
+```
+local template = grafana.template;
+ 
+dashboard.new(
+  title='Demo Dashboard'
+)
+.addTemplate(
+  template.new(
+    name='service',
+    datasource='LocalPrometheus',
+    query='label_values(service)',
+    allValues=null,
+    current='all',
+    refresh='load',
+    includeAll=true,
+    multi=true,
+  )
+)
+```
```

Comment:
I'd put the jsonnet snippet vs. equivalent generated json (or maybe an excerpt given the verbosity of it), for everyone to compare.